server/controllers/serviceSelectionValidation.ts

```diff
@@ -0,0 +1,12 @@
+import { ApiService } from '../@types/apiservice'
+
+export default class ServiceSelectionValidation {
+  static validateSelection(selections: ApiService[]): string {
```

Comment:
This isn't particularly typesafe and is prone to introduce bugs. req.body.selectedServices is either a string (if one is provided) or an array of strings (if multiple are provided).

While there's no logic in here that doesn't work for both arrays and strings, the .length below is a bit worrying purely because that's doing two different things depending on what type is passed in. I think it would be worth doing some sanitising and forcing this to be an array at the controller level, rather than letting it be passed around as either.

Reply:
Also this typing is wrong anyway - selections is not an array of ApiService objects, it's an array of string IDs.

Reply:
Added code in controller to always pass array of string for validation and added validation against available services